scripts/irods/test/test_misc.py

```diff
@@ -32,6 +34,33 @@ def test_invalid_client_irodsProt_handled_cleanly__issue_4130(self):
         out, err, ec = lib.execute_command_permissive(['ils'], env=env)
         self.assertTrue('Invalid protocol value.' in err)
 
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python', "only run for native rule language")
+    def test_acpreconnect_has_client_addr__5985(self):
+        with temporary_core_file() as core:
+            data_object_path = os.path.join(
+                self.admin.session_collection, 'importantData')
+            get_client_addr_rule = """
+                acPreConnect(*OUT) {{
+                    *OUT = "CS_NEG_DONT_CARE";
+                    msi_touch('{{"logical_path": "{path}"}}');
+                    msiModAVUMetadata("-d", "{path}", "set", "clientAddr", $clientAddr, "ip address");
+                }}
+            """.format(path=data_object_path)
+
+            time.sleep(1)  # remove once file hash fix is committed #2279
```

Comment:
I believe we could get rid of these sleep calls already, but kept them in since they were still present in other tests.

Reply:
Let's remove them from your test and see how it holds up.

Reply:
yes, #2279 is still open because we haven't removed all the `sleep()`s.   A fine thing to finish up one day...